vernac/vernacexpr.ml

```diff
@@ -343,17 +343,51 @@ type hints_expr =
   | HintsConstructors of Libnames.qualid list
   | HintsExtern of int * Constrexpr.constr_expr option * Genarg.raw_generic_argument
 
-type nonrec vernac_expr =
+type synterp_vernac_expr =
+  | VernacReservedNotation of infix_flag * (lstring * syntax_modifier CAst.t list)
+  | VernacNotation of
+      infix_flag * notation_declaration
+  | VernacDeclareCustomEntry of string
+  | VernacBeginSection of lident
+  | VernacEndSegment of lident
+  | VernacRequire of
+      qualid option * export_with_cats option * (qualid * import_filter_expr) list
+  | VernacImport of export_with_cats * (qualid * import_filter_expr) list
+  (* Modules and Module Types *)
+  | VernacDeclareModule of export_with_cats option * lident *
+      module_binder list * module_ast_inl
+  | VernacDefineModule of export_with_cats option * lident * module_binder list *
+      module_ast_inl Declaremods.module_signature * module_ast_inl list
+  | VernacDeclareModuleType of lident *
+      module_binder list * module_ast_inl list * module_ast_inl list
+  | VernacInclude of module_ast_inl list
+
+  (* Auxiliary file and library management *)
+  | VernacAddLoadPath of { implicit : bool
+                         ; physical_path : CUnix.physical_path
+                         ; logical_path : DirPath.t
+                         }
+
+  | VernacRemoveLoadPath of string
+  | VernacAddMLPath of string
+  | VernacDeclareMLModule of string list
+  | VernacChdir of string option
+  | VernacExtraDependency of qualid * string * Id.t option
+
+  | VernacSetOption of bool (* Export modifier? *) * Goptions.option_name * option_setting
+  | VernacProofMode of string
+
+  (* For extension *)
+  | VernacExtend of extend_name * Genarg.raw_generic_argument list
+
+type nonrec pure_vernac_expr =
 
   | VernacLoad of verbose_flag * string
```

Comment:
Load is not synpure.
I guess we should have 
~~~ocaml
type synterp_vernac_expr =
  | VernacLoad of verbose_flag * string
...
~~~
and something like
~~~ocaml
type synterp_entry =
  | VernacLoad of verbose_flag * vernac_entry_control list
...
~~~

Reply:
Yes, but I need to think a bit more about control there.

Reply:
Thanks for the suggestion, this is now done.